osu.Game/Overlays/MusicController.cs

```diff
@@ -389,11 +392,17 @@ public void ResetTrackAdjustments()
             CurrentTrack.RemoveAllAdjustments(AdjustableProperty.Frequency);
             CurrentTrack.RemoveAllAdjustments(AdjustableProperty.Tempo);
             CurrentTrack.RemoveAllAdjustments(AdjustableProperty.Volume);
+            audioManager.TrackMixer.Effects.Clear();
+            audioManager.SampleMixer.Effects.Clear();
```

Comment:
> - **Should local mixers be provided to ApplyTo*Mixer instead?**
I don't think any local mixers exist yet. Just providing the global mixers should be the simplest working implementation IMO.

This is percisely the reason I stated to use a local mixer. Clearing all **global** mixer effects on `ResetTrackAdjustments` is definitely bad, as that may interrupt other effects added by other components.

You can add a `DrawableAudioMixer` layer on top of the `CurrentTrack` specific for mod effects, so you can clear the effects on `ResetTrackAdjustments` safely.

Reply:
I don't know that clearing all effects from the track mixer is all that bad in the context of this method at face value (note the pre-existing and indiscriminate removal of _all_ adjustments from the current track), but touching the samples mixer as well definitely seems dodgy as the method name does not mention samples.

Reply:
In comparsion, the `RemoveAllAdjustments` call above affect the `CurrentTrack` component only, not the underlying `Beatmap.Value.Track` which certain components such as [`PlayerLoader`](https://github.com/ppy/osu/blob/98524d60a4a299ed10babd237515ca897c0b39fe/osu.Game/Screens/Play/PlayerLoader.cs#L208) and [`FailAnimation`](https://github.com/ppy/osu/blob/c6ae255326a07da5fc73ab088b4e4560715600f5/osu.Game/Screens/Play/FailAnimation.cs#L128-L129) apply adjustments to.

Meanwhile this clears all track/sample mixer effects applied on an `AudioManager` level, which is why I believe a local mixer is warranted here.

Reply:
Yeah as I said the test failures are 99% due to the effect clear so even if this is to be done as written the tests will need fixing first and foremost.

Reply:
All the test failures originate from the assertion here:
https://github.com/ppy/osu/blob/fd81842adece837b198daf8bee81960003051aa4/osu.Game/Audio/Effects/AudioFilter.cs#L129-L137

I simply removed this assertion because removing effects from the audio mixer's side sounds like a perfectly valid use case for me, and in that situation, the audio filters will still think they are attached, failing the assertion.

Also, a `DrawableAudioMixer` has been added locally in the music controller. It is a simple addition so I can get behind using a local mixer here.